### DISCUSSION TITLE
can I set the time zone?

### DISCUSSION THREAD

**Discussion by recepmrt** [thepirat000/Audit.NET#649] [Q&A]
Hi, I need to change time zone, all our systems are running according to gmt (utc) +3 and Audit logs are 3 hours behind and need to be corrected, can we do this?

thanks all.

**Comment by thepirat000** [comment]
Yes. You have the option to implement the `ISystemClock` interface that will be used to retrieve the current date and time for Audit purposes. Sadly, the property designated for getting the current date and time is called `UtcNow`; however, the returned date and time need not be strictly in UTC.

For example:

```c#
public class LocalSystemClock : ISystemClock
{
    public DateTime UtcNow => DateTime.Now; // Use the server local date, instead of the UTC
}
```

And to set-up:

```c#
Audit.Core.Configuration.SystemClock = new LocalSystemClock();
```

check this issue #227
